migration and association guides: added some remarks about join tables

r11runner opened the pull request:
Hi,

as I was reading about join tables, I had the impression that some info is contained only in the Active Record Migrations guide and some only in the Active Record Associations guide. So I tried to round out both guides a little bit.

Please review and merge, if it's OK.
Cheers,
r11runner

Commit:
[ci skip] migration and association guides: added some remarks about join tables

jonatack commented on guides/source/active_record_migrations.md:
```diff
@@ -365,9 +365,7 @@ create_join_table :products, :categories
-`false` by default. This can be overridden by specifying the `:column_options`
-option.
+`category_id` and `product_id`. (Note that the table's name is defined by the lexical order of the class names - and not by the order of the parameters.) These columns have the option `:null` set to `false` by default. This can be overridden by specifying the `:column_options` option.
 
```
It might be good to wrap lines at 80 chars like the original formatting and as per the contributing guidelines.

r11runner commented:
Thanks for the comment, I've corrected it.
Sorry, I didn't know that lines should be wrapped at 80 chars.

r11runner commented:
Added the article "the" :)

jonatack commented:
That was quick :smiley:

r11runner commented:
Punctuation and the missing spaces have been corrected in the last commit.
I also changed "will create" to "creates", because according to the API Documentation Guidelines we should prefer present tense.

claudiob commented on guides/source/active_record_migrations.md:
```diff
@@ -357,23 +357,24 @@ will append `ENGINE=BLACKHOLE` to the SQL statement used to create the table
-`false` by default. This can be overridden by specifying the `:column_options`
-option.
+`category_id` and `product_id`. (Note that the table's name is defined by the
+lexical order of the class names - and not by the order of the parameters.)
```
Hello! :grimacing: 

I think you can remove this note from this sentence and add it below.

claudiob commented on guides/source/active_record_migrations.md:
```diff
@@ -357,23 +357,24 @@ will append `ENGINE=BLACKHOLE` to the SQL statement used to create the table
+...creates the columns `product_id` and `category_id` with the `:null` option as
 `true`.
 
 You can pass the option `:table_name` when you want to customize the table
```
Add it here instead. For instance, change this sentence to:

> By default, the name of the join table comes from the union of the
> first two arguments provided to `create_join_table`, in alphabetical order.
> To customize the name of the table, provide a `:table_name` option:

jonatack replied:
Good idea.

claudiob commented on guides/source/active_record_migrations.md:
````diff
@@ -383,7 +384,7 @@ name. For example:
 ```
 
-will create a `categorization` table.
+...creates a `categorization` table.
````
I don't think the ellipses are necessary (same for line 377).

jonatack replied:
Agreed, they are not used elsewhere I the guide.

r11runner commented:
In the last commit:
 - removed the ellipses
 - changed has_and_belongs_to_many => has_and_belongs_to_many
 - added "which" to line 376
 - moved the note about the name of the join table to the section about the :table_name option

jonatack commented on guides/source/active_record_migrations.md:
```diff
@@ -367,23 +367,24 @@ create_join_table :products, :categories
+By default, the name of the join table comes from the union of the first two
+arguments provided to create_join_table, in alphabetical order.
+To customize the name of the table, provide a :table_name option:
 
```
:table_name should probably be `:table_name` here, like `:column_options` above.

r11runner commented:
Sorry, my mistake. I've corrected it in the last commit.

claudiob commented on guides/source/active_record_migrations.md:
```diff
@@ -367,23 +367,24 @@ create_join_table :products, :categories
 
-will create the `product_id` and `category_id` with the `:null` option as
-`true`.
+which creates the columns `product_id` and `category_id` with the `:null` option
```
Hello @r11runner – almost there. I think this additional paragraph is not required.
The sentence above with the example already makes it clear what happens when you specify `column_options`.
So you can just delete "which creates the columns `product_id` and `category_id` with the `:null` option as `true`."

That's all for my feedback. Please squash all your commits into one, and then your PR will be ready to get accepted.

Thanks!

r11runner commented:
@claudiob Thanks for this remark. I agree and have deleted the line "which creates the columns..."
I've squashed all commits into 1.
Let me know, if there is something else to do.

claudiob commented:
Thanks! :ribbon:
